tests/components/automation/test_time_pattern.py

```diff
@@ -23,202 +24,273 @@ def setup_comp(hass):
-        hass,
-        automation.DOMAIN,
-        {
-            automation.DOMAIN: {
-                "trigger": {
-                    "platform": "time_pattern",
-                    "hours": 1,
-                    "minutes": 2,
-                    "seconds": 3,
-                },
-                "action": {"service": "test.automation"},
-            }
-        },
+    now = dt_util.utcnow()
+    time_that_will_not_match_right_away = dt_util.utcnow().replace(
+        year=now.year + 1, hour=4
+    )
+    with patch(
+        "homeassistant.util.dt.utcnow", return_value=time_that_will_not_match_right_away
+    ):
+        assert await async_setup_component(
+            hass,
+            automation.DOMAIN,
+            {
+                automation.DOMAIN: {
+                    "trigger": {
+                        "platform": "time_pattern",
+                        "hours": 1,
+                        "minutes": 2,
+                        "seconds": 3,
+                    },
+                    "action": {"service": "test.automation"},
+                }
+            },
+        )
+
+    async_fire_time_changed(
+        hass, now.replace(year=now.year + 2, hour=1, minute=2, second=3)
     )
-
-    async_fire_time_changed(hass, dt_util.utcnow().replace(hour=1, minute=2, second=3))
 
     await hass.async_block_till_done()
     assert len(calls) == 1
 
 
 async def test_if_fires_periodic_seconds(hass, calls):
     """Test for firing periodically every second."""
-    assert await async_setup_component(
-        hass,
-        automation.DOMAIN,
-        {
-            automation.DOMAIN: {
-                "trigger": {
-                    "platform": "time_pattern",
-                    "hours": "*",
-                    "minutes": "*",
-                    "seconds": "/2",
-                },
-                "action": {"service": "test.automation"},
-            }
-        },
+    now = dt_util.utcnow()
+    time_that_will_not_match_right_away = dt_util.utcnow().replace(
+        year=now.year + 1, second=1
+    )
+    with patch(
+        "homeassistant.util.dt.utcnow", return_value=time_that_will_not_match_right_away
+    ):
+        assert await async_setup_component(
+            hass,
+            automation.DOMAIN,
+            {
+                automation.DOMAIN: {
+                    "trigger": {
+                        "platform": "time_pattern",
+                        "hours": "*",
+                        "minutes": "*",
+                        "seconds": "/10",
+                    },
+                    "action": {"service": "test.automation"},
+                }
+            },
+        )
+
+    async_fire_time_changed(
+        hass, now.replace(year=now.year + 2, hour=0, minute=0, second=10)
     )
-
-    async_fire_time_changed(hass, dt_util.utcnow().replace(hour=0, minute=0, second=2))
 
     await hass.async_block_till_done()
-    assert len(calls) == 1
+    assert len(calls) >= 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5872960)